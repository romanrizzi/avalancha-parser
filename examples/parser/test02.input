
-- Constructores
check Zero == Zero      -- sin parámetros
check Zero() == Zero()  -- sin parámetros (con paréntesis)

-- con un parámetro
check Suc(Zero) == Suc(  Zero  )

-- con varios parámetros
check A(B) == A(b)
check A(B,c) == A(b,C)
check A(B,c,d) == A(b,C,D)
check A(b,c,D,E) == A(b,C,D,E)

-- Recursivos
check A(B(C(D(E(F))))) == A(B(C(D(E(f)))))

check A(B(C),D(e,F(G))) == A(B(C),D(e,F(G)))

check Suc (
  Suc(Zero
)) == Suc(Suc(     Zero ) )

check Cons (A,Cons(B,Cons(
  C,Cons(D,
  Nil)))) == Cons(A,Cons(B,Cons(C,Cons(D,      Nil))))

