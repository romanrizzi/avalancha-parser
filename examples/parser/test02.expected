[
   "program",
   [],
   [
      [
         "check",
         [
            "equal",
            [
               "cons",
               "Zero",
               []
            ],
            [
               "cons",
               "Zero",
               []
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "cons",
               "Zero",
               []
            ],
            [
               "cons",
               "Zero",
               []
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "cons",
               "Suc",
               [
                  [
                     "cons",
                     "Zero",
                     []
                  ]
               ]
            ],
            [
               "cons",
               "Suc",
               [
                  [
                     "cons",
                     "Zero",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "cons",
               "A",
               [
                  [
                     "cons",
                     "B",
                     []
                  ]
               ]
            ],
            [
               "cons",
               "A",
               [
                  [
                     "var",
                     "b"
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "cons",
               "A",
               [
                  [
                     "cons",
                     "B",
                     []
                  ],
                  [
                     "var",
                     "c"
                  ]
               ]
            ],
            [
               "cons",
               "A",
               [
                  [
                     "var",
                     "b"
                  ],
                  [
                     "cons",
                     "C",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "cons",
               "A",
               [
                  [
                     "cons",
                     "B",
                     []
                  ],
                  [
                     "var",
                     "c"
                  ],
                  [
                     "var",
                     "d"
                  ]
               ]
            ],
            [
               "cons",
               "A",
               [
                  [
                     "var",
                     "b"
                  ],
                  [
                     "cons",
                     "C",
                     []
                  ],
                  [
                     "cons",
                     "D",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "cons",
               "A",
               [
                  [
                     "var",
                     "b"
                  ],
                  [
                     "var",
                     "c"
                  ],
                  [
                     "cons",
                     "D",
                     []
                  ],
                  [
                     "cons",
                     "E",
                     []
                  ]
               ]
            ],
            [
               "cons",
               "A",
               [
                  [
                     "var",
                     "b"
                  ],
                  [
                     "cons",
                     "C",
                     []
                  ],
                  [
                     "cons",
                     "D",
                     []
                  ],
                  [
                     "cons",
                     "E",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "cons",
               "A",
               [
                  [
                     "cons",
                     "B",
                     [
                        [
                           "cons",
                           "C",
                           [
                              [
                                 "cons",
                                 "D",
                                 [
                                    [
                                       "cons",
                                       "E",
                                       [
                                          [
                                             "cons",
                                             "F",
                                             []
                                          ]
                                       ]
                                    ]
                                 ]
                              ]
                           ]
                        ]
                     ]
                  ]
               ]
            ],
            [
               "cons",
               "A",
               [
                  [
                     "cons",
                     "B",
                     [
                        [
                           "cons",
                           "C",
                           [
                              [
                                 "cons",
                                 "D",
                                 [
                                    [
                                       "cons",
                                       "E",
                                       [
                                          [
                                             "var",
                                             "f"
                                          ]
                                       ]
                                    ]
                                 ]
                              ]
                           ]
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "cons",
               "A",
               [
                  [
                     "cons",
                     "B",
                     [
                        [
                           "cons",
                           "C",
                           []
                        ]
                     ]
                  ],
                  [
                     "cons",
                     "D",
                     [
                        [
                           "var",
                           "e"
                        ],
                        [
                           "cons",
                           "F",
                           [
                              [
                                 "cons",
                                 "G",
                                 []
                              ]
                           ]
                        ]
                     ]
                  ]
               ]
            ],
            [
               "cons",
               "A",
               [
                  [
                     "cons",
                     "B",
                     [
                        [
                           "cons",
                           "C",
                           []
                        ]
                     ]
                  ],
                  [
                     "cons",
                     "D",
                     [
                        [
                           "var",
                           "e"
                        ],
                        [
                           "cons",
                           "F",
                           [
                              [
                                 "cons",
                                 "G",
                                 []
                              ]
                           ]
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "cons",
               "Suc",
               [
                  [
                     "cons",
                     "Suc",
                     [
                        [
                           "cons",
                           "Zero",
                           []
                        ]
                     ]
                  ]
               ]
            ],
            [
               "cons",
               "Suc",
               [
                  [
                     "cons",
                     "Suc",
                     [
                        [
                           "cons",
                           "Zero",
                           []
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "cons",
               "Cons",
               [
                  [
                     "cons",
                     "A",
                     []
                  ],
                  [
                     "cons",
                     "Cons",
                     [
                        [
                           "cons",
                           "B",
                           []
                        ],
                        [
                           "cons",
                           "Cons",
                           [
                              [
                                 "cons",
                                 "C",
                                 []
                              ],
                              [
                                 "cons",
                                 "Cons",
                                 [
                                    [
                                       "cons",
                                       "D",
                                       []
                                    ],
                                    [
                                       "cons",
                                       "Nil",
                                       []
                                    ]
                                 ]
                              ]
                           ]
                        ]
                     ]
                  ]
               ]
            ],
            [
               "cons",
               "Cons",
               [
                  [
                     "cons",
                     "A",
                     []
                  ],
                  [
                     "cons",
                     "Cons",
                     [
                        [
                           "cons",
                           "B",
                           []
                        ],
                        [
                           "cons",
                           "Cons",
                           [
                              [
                                 "cons",
                                 "C",
                                 []
                              ],
                              [
                                 "cons",
                                 "Cons",
                                 [
                                    [
                                       "cons",
                                       "D",
                                       []
                                    ],
                                    [
                                       "cons",
                                       "Nil",
                                       []
                                    ]
                                 ]
                              ]
                           ]
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ]
   ]
]
